addons/spreadsheet_account/static/tests/model/partner_balance.js

```diff
@@ -0,0 +1,85 @@
+import { setCellContent } from "@spreadsheet/../tests/utils/commands";
+import {
+    createModelWithDataSource,
+    waitForDataSourcesLoaded,
+} from "@spreadsheet/../tests/utils/model";
+import { getEvaluatedCell } from "@spreadsheet/../tests/utils/getters";
+import "@spreadsheet_account/index";
+
+QUnit.module("spreadsheet_account > partner balance", {}, () => {
+    QUnit.test("Basic evaluation", async (assert) => {
+        const model = await createModelWithDataSource({
+            mockRPC: async function (route, args) {
+                if (args.method === "get_partner_balance") {
+                    assert.step("get_partner_balance");
+                    assert.deepEqual(args.args, [
+                        [
+                            {
+                                partner_ids: [
+                                    14, 16
+                                ],
+                                codes: [
+                                    "112",
+                                ],
+                                date_range: {
+                                    range_type: "year",
+                                    year: 2023,
+                                },
+                                company_id: 0,
+                                include_unposted: false,
+                            }
+                        ],
+                    ]);
+                    return [111.11];
+                }
+            },
+        });
+        setCellContent(model, "A1", `=ODOO.PARTNER.BALANCE("14, 16", "112", 2023)`);
+        await waitForDataSourcesLoaded(model);
+        assert.verifySteps(["get_partner_balance"]);
+        assert.equal(getEvaluatedCell(model, "A1").value, "111.11");
+    });
+
+    QUnit.test("with wrong date format", async (assert) => {
+        const model = await createModelWithDataSource();
+        setCellContent(model, "A1", `=ODOO.PARTNER.BALANCE("14, 16", "112", "This is a valid date")`);
```

Comment:
did you mean `not a valid date`?
```suggestion
        setCellContent(model, "A1", `=ODOO.PARTNER.BALANCE("14, 16", "112", "This is not a valid date")`);
```